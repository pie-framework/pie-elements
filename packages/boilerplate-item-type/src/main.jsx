import React from 'react';
import PropTypes from 'prop-types';
import { PreviewPrompt } from '@pie-element/pie-utils/render-ui';
import UTILS from '@pie-element/pie-utils';

const {test, callTest} = UTILS;

callTest('player');
export default class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func.isRequired,
  };

  static defaultProps = {};

  render() {
    const { model } = this.props;

    return (
      <div>
        {model.prompt && (
          <div>
            <p>{test}</p>
            <p>Boilerplate Item</p>
            <PreviewPrompt prompt={model.prompt}/>
          </div>
        )}
      </div>
    );
  }
}
