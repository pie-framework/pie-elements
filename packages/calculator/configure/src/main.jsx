import React from 'react';
import PropTypes from 'prop-types';
import { layout, TwoChoice } from '@pie-lib/config-ui';
import { CalculatorLayout } from '@pie-element/calculator';
import Typography from '@material-ui/core/Typography';

const MainTypes = {
  model: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
};

export class Main extends React.Component {
  static propTypes = MainTypes;

  onModeChange = (mode) => {
    const { model, onChange } = this.props;
    const update = { ...model, mode };

    onChange(update);
  };

  render() {
    const { model } = this.props;

    return (
      <layout.ConfigLayout hideSettings={true} settings={null}>
        <TwoChoice
          header="Choose a Calculator"
          value={model.mode}
          onChange={this.onModeChange}
          one={{ label: 'Basic', value: 'basic' }}
          two={{ label: 'Scientific', value: 'scientific' }}
        />
        <br />

        <CalculatorLayout mode={model.mode} onClose={() => ({})} />
        <br />

        <Typography style={{ paddingTop: '4px' }}>
          Please note that the calculators are tools for students and do not record answers.
        </Typography>
      </layout.ConfigLayout>
    );
  }
}

export default class Stateful extends React.Component {
  static propTypes = MainTypes;

  constructor(props) {
    super(props);
    this.state = { model: props.model };
  }

  UNSAFE_componentWillReceiveProps(props) {
    this.setState({ model: props.model });
  }

  onChange = (model) => {
    this.setState({ model }, () => {
      this.props.onChange(this.state.model);
    });
  };

  render() {
    return <Main model={this.state.model} onChange={this.onChange} />;
  }
}
