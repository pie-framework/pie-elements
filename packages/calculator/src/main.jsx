import React from 'react';
import PropTypes from 'prop-types';
import DraggableCalculator from './draggable-calculator';
import CalculatorIcon from './calculator-icon';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';

const styles = {
  icon: {
    transition: 'fill 200ms',
    '&:hover': {
      fill: 'black',
    },
    cursor: 'pointer',
    verticalAlign: 'middle',
    fill: 'grey',
  },
  active: {
    fill: 'black',
  },
};

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
  };
  constructor(props) {
    super(props);
    this.state = { show: false };
  }

  onToggleShow = () => this.setState({ show: !this.state.show });
  onClickClose = () => this.setState({ show: false });

  render() {
    const { show } = this.state;
    const { mode } = this.props.model;
    const { classes } = this.props;
    return (
      <div>
        <CalculatorIcon
          className={show ? classNames(classes.icon, classes.active) : classes.icon}
          onClick={() => this.onToggleShow()}
        />
        <DraggableCalculator mode={mode} show={show} onClose={this.onClickClose} />
      </div>
    );
  }
}

export default withStyles(styles)(Main);
