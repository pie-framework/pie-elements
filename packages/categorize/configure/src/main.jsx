import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Design from './design';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    model: PropTypes.object.isRequired,
    onModelChanged: PropTypes.func.isRequired,
    imageSupport: PropTypes.object
  };

  static defaultProps = {};

  render() {
    const {
      classes,
      className,
      model,
      onModelChanged,
      configuration,
      onConfigurationChanged
    } = this.props;

    return (
      <div className={classNames(classes.main, className)}>
        <Design
          imageSupport={this.props.imageSupport}
          title="Design"
          model={model}
          configuration={configuration}
          onChange={onModelChanged}
          onConfigurationChanged={onConfigurationChanged}
        />
      </div>
    );
  }
}
const styles = theme => ({
  main: {}
});

export default withStyles(styles)(Main);
