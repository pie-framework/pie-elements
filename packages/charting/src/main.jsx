import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { color, Collapsible, hasText } from '@pie-lib/render-ui';
import { Chart, chartTypes } from '@pie-lib/charting';
import isEqual from 'lodash/isEqual';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import {GraphContainer} from '@pie-lib/graphing';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    model: PropTypes.object.isRequired,
    onAnswersChange: PropTypes.func,
    categories: PropTypes.array,
  };

  static defaultProps = { classes: {} };

  constructor(props) {
    super(props);

    this.state = {
      categories: props.categories || props.model.data,
      showingCorrect: false
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { model: { data: nextData = [] } = {} } = nextProps;
    const { model: { data = [] } = {} } = this.props;

    if (!isEqual(nextData, data)) {
      this.setState({ categories: nextData });
    }
  }

  changeData = (data) =>
    this.setState(
      {
        categories: data,
      },
      () => this.props.onAnswersChange(data)
    );

  toggleCorrect = showingCorrect => this.setState({ showingCorrect });

  render() {
    const { categories, showingCorrect } = this.state;
    const { model, classes } = this.props;
    const {
      teacherInstructions,
      prompt,
      chartType,
      size,
      domain,
      range,
      title,
      editCategoryEnabled,
      addCategoryEnabled,
      categoryDefaultLabel,
      rationale,
      correctedAnswer,
      correctAnswer
    } = model;

    const correctData = correctAnswer && correctAnswer.data ? correctAnswer.data.map(data => {
      return {
        ...data,
        interactive: false,
        editable: false
      };
    }) : [];

    const showToggle = correctData && correctData.length > 0;

    return (
      <div className={classes.mainContainer}>
        <CorrectAnswerToggle
          show={showToggle}
          toggled={showingCorrect}
          onToggle={this.toggleCorrect}
        />

        {teacherInstructions && hasText(teacherInstructions) && (
          <React.Fragment>
            <Collapsible
              labels={{
                hidden: 'Show Teacher Instructions',
                visible: 'Hide Teacher Instructions',
              }}
            >
              <div dangerouslySetInnerHTML={{ __html: teacherInstructions }} />
            </Collapsible>
            <br />
          </React.Fragment>
        )}

        {prompt && (
          <React.Fragment>
            <div
              className={classes.prompt}
              dangerouslySetInnerHTML={{ __html: prompt }}
            />
            <br />
          </React.Fragment>
        )}

        {(showingCorrect && showToggle) ? (
          <Chart
            chartType={chartType}
            size={size}
            domain={domain}
            range={range}
            charts={[
              chartTypes.Bar(),
              chartTypes.Histogram(),
              chartTypes.LineDot(),
              chartTypes.LineCross(),
              chartTypes.DotPlot(),
              chartTypes.LinePlot(),
            ]}
            data={correctData || categories}
            title={title}
            onDataChange={this.changeData}
            editCategoryEnabled={false}
            addCategoryEnabled={false}
            categoryDefaultLabel={categoryDefaultLabel}
          />
        ) : (
          <Chart
            chartType={chartType}
            size={size}
            domain={domain}
            range={range}
            charts={[
              chartTypes.Bar(),
              chartTypes.Histogram(),
              chartTypes.LineDot(),
              chartTypes.LineCross(),
              chartTypes.DotPlot(),
              chartTypes.LinePlot(),
            ]}
            data={correctedAnswer || categories}
            title={title}
            onDataChange={this.changeData}
            editCategoryEnabled={editCategoryEnabled}
            addCategoryEnabled={addCategoryEnabled}
            categoryDefaultLabel={categoryDefaultLabel}
          />)}

        <br />
        {rationale && hasText(rationale) && (
          <Collapsible
            labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}
          >
            <div dangerouslySetInnerHTML={{ __html: rationale }} />
          </Collapsible>
        )}
      </div>
    );
  }
}

const styles = theme => ({
  mainContainer: {
    padding: theme.spacing.unit,
    color: color.text(),
    backgroundColor: color.background(),
  },
  prompt: {
    verticalAlign: 'middle',
  },
});

export default withStyles(styles)(Main);
