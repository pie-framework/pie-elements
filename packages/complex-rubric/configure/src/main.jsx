import React from 'react';
import PropTypes from 'prop-types';
import { RUBRIC_TYPES } from '@pie-lib/pie-toolbox/rubric';
import { layout } from '@pie-lib/pie-toolbox/config-ui';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import { withStyles } from '@material-ui/core/styles';
import { FormControlLabel } from '@material-ui/core';

const styles = {};
const rubricLabels = {
  [RUBRIC_TYPES.MULTI_TRAIT_RUBRIC]: 'Multi Trait Rubric',
  [RUBRIC_TYPES.SIMPLE_RUBRIC]: 'Simple Rubric',
  [RUBRIC_TYPES.RUBRICLESS]: 'Rubricless',
};

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    canUpdateModel: PropTypes.bool,
    configuration: PropTypes.object,
    model: PropTypes.object,
    onModelChanged: PropTypes.func,
    onConfigurationChanged: PropTypes.func,
  };

  onModelChanged = (model) => {
    const { onModelChanged } = this.props;

    return onModelChanged(model);
  };

  onChangeRubricType = (e) => {
    const { model } = this.props;

    const rubricType = e.target.value;
    this.onModelChanged({ ...model, rubricType });
  };

  render() {
    const { model, configuration, canUpdateModel } = this.props;

    const { rubrics = {} } = model || {};
    let { rubricType } = model;
    const { contentDimensions = {}, rubricOptions = [], multiTraitRubric, simpleRubric, rubricless, width } = configuration;
    let rubricTag = '';

    if (!rubricType) {
      rubricType = RUBRIC_TYPES.SIMPLE_RUBRIC;
    }

    const isRubricTypeAvailable =  rubricOptions.indexOf(rubricType) > -1;

    if (canUpdateModel) {
      switch (rubricType) {
        case RUBRIC_TYPES.SIMPLE_RUBRIC:
        default:
          rubricTag = (
            <rubric-configure
              id="simpleRubric"
              key="simple-rubric"
              ref={(ref) => {
                if (ref) {
                  this.simpleRubric = ref;
                  this.simpleRubric.model = {...rubrics.simpleRubric, errors: model.errors || {}};
                  this.simpleRubric.configuration = { ...simpleRubric, width };
                }
              }}
            />
          );
          break;

        case RUBRIC_TYPES.MULTI_TRAIT_RUBRIC:
          rubricTag = (
            <multi-trait-rubric-configure
              id="multiTraitRubric"
              key="multi-trait-rubric"
              ref={(ref) => {
                if (ref) {
                  this.multiTraitRubric = ref;

                  this.multiTraitRubric.model = {...rubrics.multiTraitRubric, errors: model.errors || {} };
                  this.multiTraitRubric.configuration = { ...multiTraitRubric, width: width || multiTraitRubric.width };
                }
              }}
            />
          );
          break;

        case RUBRIC_TYPES.RUBRICLESS:
          rubricTag = (
              <rubric-configure
                  id="rubricless"
                  key="rubricless"
                  ref={(ref) => {
                    if (ref) {
                      this.rubricless = ref;

                      this.rubricless.model = rubrics.rubricless;
                      this.rubricless.configuration = { ...rubricless, width };
                    }
                  }}
              />
          );
          break;
      }
    }

    return (
      <layout.ConfigLayout dimensions={contentDimensions} hideSettings={true} settings={null}>
        <RadioGroup
          aria-label="rubric-type"
          name="rubricType"
          value={model.rubricType}
          onChange={this.onChangeRubricType}
        >
          {
            rubricOptions.length > 1 && rubricOptions.map((availableRubric, i)=>
              <FormControlLabel
                key={i}
                value={availableRubric}
                control={<Radio checked={rubricType === availableRubric} />}
                label={rubricLabels[availableRubric]}
            />
          )
          }
        </RadioGroup>

        {isRubricTypeAvailable && rubricTag}
      </layout.ConfigLayout>
    );
  }
}

export default withStyles(styles)(Main);
