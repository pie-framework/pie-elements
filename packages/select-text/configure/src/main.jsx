import React from 'react';
import PropTypes from 'prop-types';
import Design from './design';
import { withStyles } from '@material-ui/core/styles';
import cloneDeep from 'lodash/cloneDeep';

export class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  static defaultProps = {};

  onPromptChanged = prompt => {
    const { onChange, model } = this.props;
    const update = cloneDeep(model);

    update.prompt = prompt;
    onChange(update);
  };

  render() {
    const { model, onChange, classes, imageSupport } = this.props;

    return (
      <Design
        className={classes.container}
        title={'Design'}
        model={model}
        onChange={onChange}
        imageSupport={imageSupport}
        onPromptChanged={this.onPromptChanged}
      />
    );
  }
}
export default withStyles(theme => ({
  container: {
    paddingTop: theme.spacing.unit
  }
}))(Main);
