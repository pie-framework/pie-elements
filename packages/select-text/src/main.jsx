import React from 'react';
import PropTypes from 'prop-types';
import { TextSelect, Legend } from '@pie-lib/text-select';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { color, Feedback, Collapsible, hasText, PreviewPrompt } from '@pie-lib/render-ui';
import { withStyles } from '@material-ui/core/styles';
import generateModel from './utils';

import debug from 'debug';

const log = debug('@pie-ui:select-text');

const Types = {
  model: PropTypes.object,
  session: PropTypes.object,
  onSelectionChange: PropTypes.func.isRequired,
};

export class Main extends React.Component {
  static propTypes = { ...Types, classes: PropTypes.object.isRequired };

  static defaultProps = {};

  constructor(props) {
    super(props);

    this.state = {
      showCorrectAnswer: this.props.model.alwaysShowCorrect || false,
      model: generateModel(props.model),
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({
      showCorrectAnswer: !!nextProps.model.alwaysShowCorrect,
      model: generateModel(nextProps.model),
    });
  }

  toggleShowCorrect = () => {
    this.setState({ showCorrectAnswer: !this.state.showCorrectAnswer });
  };

  correctAnswer = () => {
    const { model } = this.state;

    return model.tokens.filter((t) => t.correct);
  };

  render() {
    const { session, onSelectionChange, classes } = this.props;
    const { showCorrectAnswer, model } = this.state;
    const { env } = model;
    const { mode } = env || {};


    const selectedTokens = showCorrectAnswer ? this.correctAnswer() : session.selectedTokens;

    log('[render] selectedTokens:', selectedTokens);

    return (
      <div className={classes.mainContainer}>
        {model.teacherInstructions &&
          hasText(model.teacherInstructions) &&
          (!model.animationsDisabled ? (
            <Collapsible
              labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
              className={classes.collapsible}
            >
              <PreviewPrompt prompt={model.teacherInstructions} />
            </Collapsible>
          ) : (
            <div className={classes.prompt}>
              <PreviewPrompt prompt={model.teacherInstructions} />
            </div>
          ))}

        <div className={classes.prompt}>
          <PreviewPrompt prompt={model.prompt} />
        </div>

        {!model.alwaysShowCorrect && (
          <CorrectAnswerToggle
            show={model.disabled && model.incorrect}
            toggled={showCorrectAnswer}
            onToggle={this.toggleShowCorrect}
            language={model.language}
          />
        )}

        <TextSelect
          className={classes.textSelect}
          disabled={model.disabled}
          text={model.text}
          tokens={model.tokens}
          selectedTokens={selectedTokens}
          onChange={(selection) => {
            const newSelections = selection.map((select) => {
              const token = model.tokens.find(({ start, end }) => select.start === start && select.end === end);

              // needed for getScore when tokens position is recalculated, to keep oldStart and oldEnd
              if (token) {
                return token;
              }

              return select;
            });

            onSelectionChange(newSelections);
          }}
          highlightChoices={model.highlightChoices}
          maxNoOfSelections={model.maxSelections}
          animationsDisabled={model.animationsDisabled}
        />
        { mode === 'evaluate' && <Legend></Legend> }

        {model.rationale &&
          hasText(model.rationale) &&
          (!model.animationsDisabled ? (
            <Collapsible
              labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}
              className={classes.collapsible}
            >
              <PreviewPrompt prompt={model.rationale} />
            </Collapsible>
          ) : (
            <div className={classes.prompt}>
              <PreviewPrompt prompt={model.rationale} />
            </div>
          ))}

        {model.correctness && model.feedback && !showCorrectAnswer && (
          <Feedback correctness={model.correctness} feedback={model.feedback} />
        )}
      </div>
    );
  }
}

const StyledMain = withStyles((theme) => ({
  mainContainer: {
    color: color.text(),
    backgroundColor: color.background(),
  },
  textSelect: {
    marginBottom: theme.spacing.unit,
    marginTop: theme.spacing.unit * 2,
    whiteSpace: 'normal',
  },
  prompt: {
    verticalAlign: 'middle',
    marginBottom: theme.spacing.unit,
  },
  collapsible: {
    marginBottom: theme.spacing.unit * 2,
  },
}))(Main);

export default class Stateful extends React.Component {
  static propTypes = Types;

  constructor(props) {
    super(props);
    this.state = {
      model: props.model,
      session: props.session,
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ model: nextProps.model, session: nextProps.session });
  }

  change = (selectedTokens) => {
    const session = { ...this.state.session, selectedTokens };
    this.setState({ session }, () => {
      this.props.onSelectionChange(this.state.session.selectedTokens);
    });
  };

  render() {
    const { model, session } = this.state;
    return <StyledMain model={model} session={session} onSelectionChange={this.change} />;
  }
}
