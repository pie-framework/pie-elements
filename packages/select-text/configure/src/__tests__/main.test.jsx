import { shallow } from 'enzyme';
import React from 'react';
import { Main } from '../main';
import {
  layout,
  settings,
} from '@pie-lib/config-ui';

jest.mock('@pie-lib/config-ui', () => ({
  layout: {
    ConfigLayout: props => <div>{props.children}</div>
  },
  settings: {
    Panel: props => <div onChange={props.onChange} />,
    toggle: jest.fn(),
    radio: jest.fn()
  }
}));


describe('main', () => {
  let w;
  let onChange;

  const getModel = () => ({ tokens: [] });
  beforeEach(() => {
    onChange = jest.fn();
    w = shallow(<Main model={getModel()} classes={{}} onChange={onChange} />);
  });

  describe('snapshot', () => {
    it('renders', () => {
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {});
});
