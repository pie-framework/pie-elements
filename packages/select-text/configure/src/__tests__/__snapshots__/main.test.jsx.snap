// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main snapshot renders 1`] = `
<WithStyles(Design)
  model={
    Object {
      "tokens": Array [],
    }
  }
  onChange={[MockFunction]}
  onPromptChanged={[Function]}
  title="Design"
/>
`;
